svg-slideshow.py

```diff
@@ -1,64 +1,77 @@
 #!/usr/bin/env python3
 
 import imghdr
-
+import argparse
 import base64
-import sys
-
-args = sys.argv[1:]
-
-frames = len(args)
-
-print("Trying to load %d images" % frames)
-
-width = 800
-height = 500
-fade = 500 #ms
-delay = 2000 #ms
-
-dur = (fade + delay) * frames
-print("Slideshow will be %d ms" % dur)
-
-f = open('out.svg','w')
-
-f.write('<?xml version="1.0" encoding="utf-8"?>\n')
-f.write('<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="%d" height="%d" viewBox="0 0 %d %d">\n' % (width, height, width, height))
-
-t = 0
-
-def startImage(path):
-  with open(path, "rb") as image_file:
-    buf = image_file.read()
-    fmt = imghdr.what(None, buf)
-    encoded = base64.b64encode(buf)
-  f.write('<image opacity="1.0" width="%d" height="%d" xlink:href="data:image/%s;base64,%s">\n' % (width, height, fmt, encoded.decode("utf-8")))
-
-args = args[1:] + [args[0]]
-
-startImage(args[-1])
-f.write('</image>\n')
-
-for p in args:
-
-#    <!-- Fallback if animate is not supported -->
+from xml.etree import ElementTree as tree
+from PIL import Image
+
+
+def init_parser():
+    parser = argparse.ArgumentParser(description="Utility to generate animated SVGs for a slideshow from images")
+    parser.add_argument('files', nargs='+')
+    parser.add_argument('-x', '--width', type=int, default=800, help='output image width, default is 800')
+    parser.add_argument('-y', '--height', type=int, default=500, help='output image height, default is 800')
+    parser.add_argument('-f', '--fade', type=int, default=500, help='duration of fade in ms')
+    parser.add_argument('-d', '--delay', type=int, default=2000, help='delay in ms')
```

Comment:
The default for these 2 isn't stated. I'd recommend doing it for all, or none.
Also this could be more descriptive like "duration how long each image is shown between fades in ms". Otherwise it could be mistaken for an initial delay before anything happens (which isn't a feature yet).